src/ejabberd_c2s.erl

```diff
@@ -3364,3 +3361,12 @@ maybe_hibernate(#state{hibernate_after = HA}) -> HA.
 
 make_c2s_info(_StateData = #state{stream_mgmt_buffer_size = SMBufSize}) ->
     #{stream_mgmt_buffer_size => SMBufSize}.
+
+-spec update_stanza(jid:jid(), jid:jid(), exml:element(), mongoose_acc:t()) ->
+    mongoose_acc:t().
+update_stanza(From, To, #xmlel{} = Element, Acc) ->
+    LServer = mongoose_acc:lserver(Acc),
+    Params = #{lserver => LServer, element => Element,
+               from_jid => From, to_jid => To},
+    NewAcc = mongoose_acc:strip(Params, Acc),
+    mongoose_acc:delete_many(c2s, [origin_jid, origin_sid], NewAcc).
```

Comment:
Why do we delete these?

Reply:
other way another C2S drops some of the bounced messages. see sm:subscription_requests_are_buffered_properly testcase

Reply:
added TODO comment to make further investigations later.